test/unit/segment-spec.js

```diff
@@ -1,59 +1,141 @@
 'use strict';
 
+var Peaks = require('../../src/main');
 var Segment = require('../../src/main/markers/segment');
 
 describe('Segment', function() {
+  var p, deprecationLogger;
+
+  beforeEach(function(done) {
+    deprecationLogger = sinon.spy();
+    p = Peaks.init({
+      container: document.getElementById('waveform-visualiser-container'),
+      mediaElement: document.querySelector('audio'),
+      dataUri: {
+        json: 'base/test_data/sample.json'
+      },
+      keyboard: true,
+      height: 240,
+      deprecationLogger: deprecationLogger
+    });
+    p.on('peaks.ready', done);
+  });
+
+  afterEach(function() {
+    if (p) {
+      p.destroy();
+    }
+  });
+
+  describe('labelText', function() {
+    it('should be possible to update programatically', function() {
+      p.segments.add({ startTime: 0, endTime: 10, labelText: 'label text' });
+
+      var newLabelText = 'new label text';
+      var segment = p.segments.getSegments()[0];
+
+      segment.labelText = newLabelText;
+      expect(segment.labelText).to.equal(newLabelText);
+    });
+  });
+
+  describe('startTime', function() {
+    it('should be possible to update programatically', function() {
```

Comment:
I just realised an issue with changing the start and end times. If you have a segment, for example at (5, 10) and you want to move it to (15, 20), you have to update the end time first, otherwise you get an exception. I wonder if we should prefer to have an `updateTime(start, end)` method to be able to set both together? Or maybe to flip the start and end times automatically if you set the end time after the start time?

Reply:
ah okay. I think it would make sense if the startTime/endTime were always updated together in this case. So NOT providing a means of updating starTime and endTime, and instead providing a setter function `updateTime(start, end)`?

Reply:
This should be resolved now

Reply:
I wonder if the API would be simpler if we added a `segment.update()` method, that takes an object with the values you want to change, rather than use getter and setters. For example:

```javascript
segment.update({ labelText: 'New label' });
segment.update({ labelText, 'New label', startTime: 20, endTime: 30 });
```

What do you think?

Reply:
I agree that is a better approach

Reply:
This is added in commit #6f32044

Reply:
Ah yes, we don't currently use any ES6 features... although I would like to modernise the code at some point.